### PR TITLE
don't do extra  outgoing p2p-rta connections

### PR DESCRIPTION
**Pull request by mbg033**

### CODE REVIEW

File: src/p2p/net_node.inl [hunk 1]
```diff
@@ -1671,6 +1673,27 @@ namespace nodetool
     });
     return ret;
   }
+
+  //-----------------------------------------------------------------------------------
+  template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::find_connection_id_by_peer(const peerlist_entry &pe, boost::uuids::uuid& conn_id)
```
Comment (bitkis) [inline src/p2p/net_node.inl]: 1. Any chance we're duplicating code here? I have feeling I saw this before
2. explicit capturing please

File: src/p2p/net_node.inl [hunk 2]
```diff
@@ -2297,10 +2320,11 @@ namespace nodetool
     // send to peers
     m_net_server.get_config_object().foreach_connection([&](p2p_connection_context& context) {
         LOG_INFO_CC(context, "invoking COMMAND_SUPERNODE_ANNOUNCE");
-        if (context.peer_id == 0) {
-            LOG_INFO_CC(context, "invalid connection [COMMAND_SUPERNODE_ANNOUNCE]");
-            return true;
-        }
+        // XXX: peer_id is always 0 for RTA peers. it only generated properly as the part of handshake procedure
```
Comment (bitkis) [inline src/p2p/net_node.inl]: do we need this code?

File: src/p2p/net_node.inl (hunk 1 above)
Comment (mbg033) [reply src/p2p/net_node.inl]: 1 - deleted old function;
2 - done;

Comment (mbg033) [reply src/p2p/net_node.inl]: squashed into one commit

File: src/p2p/net_node.inl (hunk 2 above)
Comment (mbg033) [reply src/p2p/net_node.inl]: deleted

**Pull request merged**